src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs

```diff
@@ -106,13 +106,14 @@ protected async override Task<UpgradeStepInitializeResult> InitializeImplAsync(I
             }
 
             var componentFlagProperty = context.Properties.GetPropertyValue("componentFlag");
-            if (componentFlagProperty is null)
-            {
-                return new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, "componentFlag Property in Context was null", BuildBreakRisk.High);
-            }
 
             // This block checks TFMs for .NET MAUI platform projects
             var targetTfm = GetExpectedTargetFramework(componentFlagProperty);
+            if (targetTfm is null)
+            {
+                return new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"targetTfm or componentFlag Property in Context was null.", BuildBreakRisk.High);
+            }
```

Comment:
Let's make the failure message consistent between these two places and include the property value:

$"Failed to retrieve target TFM from component flag context property: '{componentFlagProperty}'"